### PR TITLE
US-7 Event Distance

### PR DESCRIPTION
**Pull request by toshydev**
- add getDistance function, add eventsInRange filter
- add distance calculation to EventPreview and EventDetail
- add error handling for currentLocation false

### CODE REVIEW

File: src/utils/getEvents.js [hunk 1]
```diff
@@ -16,3 +18,17 @@ export function getEventsOfCity(city, events) {
...
+export function getEventsInRange(userCoords, range, events) {
+  return events.filter((event) => {
+    return (
+      getDistance(
+        { latitude: userCoords[0], longitude: userCoords[1] },
+        {
+          latitude: event.location.coordinates[0],
+          longitude: event.location.coordinates[1],
+        }
+      ) <= range
```
Comment (augustinanders) [inline src/utils/getEvents.js:31]: maybe it is because i dont know the libary, but what is the backwards arrow for here?

**Review by augustinanders — APPROVED** [review]
There were  many code snippets changed and i didn't understand every detail, but LGTM in general.

File: src/utils/getEvents.js (hunk 1 above)
Comment (toshydev) [reply src/utils/getEvents.js:31]: It's the lesser than operator. It's looks confusing because of the line break.

Comment (toshydev) [reply src/utils/getEvents.js:31]: The library has useful formulas e.g. for calculating the distance between two coordinates.

**Review by rabe200 — APPROVED** [review]
also had to read a lot of new docs because of unknown librariers. but very interesting. lgtm aswell!

**Pull request merged**